azurelinuxagent/ga/firewall_manager.py

```diff
@@ -0,0 +1,499 @@
+        """
+        Returns the command to manipulate the rule for dropping all requests on the WireServer address.
+        """
+        raise NotImplementedError()
+
+    def _get_legacy_rule_command(self, command_option):
+        """
+        Returns the command to delete the legacy firewall rule.
+
+        See the overrides of this method for details on those rules.
+        """
+        raise NotImplementedError()
+
+    def _get_append_command_option(self):
+        """
+        Returns the command-line option to append a firewall to the output chain.
+        """
+        raise NotImplementedError()
+
+    def _get_check_command_option(self):
+        """
+        Returns the command-line option to check for existence of rule on the output chain.
+        """
+        raise NotImplementedError()
+
+    def _get_delete_command_option(self):
+        """
+        Returns the command-line option to delete a firewall rule from the output chain.
+        """
+        raise NotImplementedError()
+
+
+class IpTables(_FirewallManagerMultipleRules):
+    """
+    FirewallManager based on the iptables command-line tool.
+    """
+    def __init__(self, wire_server_address):
+        super(IpTables, self).__init__(wire_server_address)
+
+        #
+        # The wait option, "-w" was introduced in iptables 1.4.21. Check if we can use it.
+        #
+        try:
+            output = shellutil.run_command(["iptables", "--version"])
+            #
+            # The output is similar to
+            #
+            #     $ iptables --version
+            #     iptables v1.8.7 (nf_tables)
+            #
+            # Extract anything that looks like a version number.
+            #
+            match = re.match(r"^[^\d.]*([\d.]+).*$", output)
+            if match is None:
+                raise Exception('output of "--version": {0}'.format(output))
+            version = FlexibleVersion(match.group(1))
+            use_wait_option = version >= FlexibleVersion('1.4.21')
+
+        except Exception as exception:
+            if isinstance(exception, OSError) and exception.errno == errno.ENOENT:  # pylint: disable=no-member
+                raise FirewallManagerNotAvailableError("iptables is not available")
+            message = "Unable to determine version of iptables; will not use -w option. --version output: {0}".format(ustr(exception))
+            logger.warn(message)
+            add_event(op=WALAEventOperation.Firewall, is_success=False, message=message, log_event=False)
+            use_wait_option = False
+
+        if use_wait_option:
+            self._base_command = ["iptables", "-w", "-t", "security"]
+        else:
+            self._base_command = ["iptables", "-t", "security"]
+
+    def _execute_delete_command(self, command):
+        """
+        Continually execute the delete operation until the return code is non-zero or the limit has been reached.
+        """
+        for _ in range(1, 100):
```

Comment:
Yes, same thing here. I have no idea why this written this way; might be related to the -w option (for those versions of iptables that do not have it). If time allows, I will look more into this for this release, otherwise will do on a subsequent release.